src/App.js

```diff
@@ -1,34 +1,36 @@
-import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
-import Layout from './components/Layout'
+import { Router, Switch, Route } from 'react-router-dom';
+import PrivateRoute from './utils/PrivateRoute';
+import Layout from './components/Layout';
 import LoginPage from './pages/LoginPage';
 import Register from './pages/Register';
 import { LandingPage } from './pages/LandingPage';
-
 import { useSelector } from 'react-redux';
 import { useDispatch } from 'react-redux';
-
 import getUserData from './actions/getUserData';
 import { useEffect } from 'react';
-import { errorPage } from './pages/errorPage'
+import { errorPage } from './pages/errorPage';
 import HomePage from './pages/HomePage';
+import history from './utils/history';
 
 function App() {
-  const state = useSelector(state => state)
+  const state = useSelector((state) => state);
```

Comment:
Please use state key to prevent rerendenring. in this case use state.token (That only render page when token is updated )

Reply:
Hi Leonel, just refactored the code and changed the state constant for the specific global state token